c7n/resources/secretsmanager.py

```
@@ -0,0 +1,136 @@
+                if e['Error']['Code'] == 'InternalFailure':
+                    self.log.error(
+                        'Exception when tagging resource %s' %
+                        (r['ARN'], e['Error']['Code']))
+
+
+@SecretsManager.action_registry.register('remove-tag')
+class RemoveTagSecretsManagerResource(RemoveTag):
+    """Action to remove tag(s) on a Secret resource
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: untag-secret
+              resource: secrets-manager
+              actions:
+                - type: remove-tag
+                  tags: ['tag-to-be-removed']
+    """
+
+    permissions = ('secretsmanager:UntagResource',)
+
+    def process_resource_set(self, resources, keys):
+        client = local_session(self.manager.session_factory).client('secretsmanager')
+        for r in resources:
+            client.untag_resource(SecretId=r['ARN'], TagKeys=keys)
+
+
+@SecretsManager.action_registry.register('mark-for-op')
+class MarkSecretForOp(TagDelayedAction):
+    """Action to mark a Secret resource for deferred action :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: mark-secret-for-delete
+              resource: secrets-manager
+              actions:
+                - type: mark-for-op
+                  op: delete
+                  days: 1
+    """
+
+    permissions = ('secretsmanager:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('secretsmanager')
+        for r in resources:
+            r['Tags'].extend([{'Key': t['Key'], 'Value':t['Value']} for t in tags])
+            try:
+                client.tag_resource(SecretId=r['ARN'], Tags=r['Tags'])
+            except ClientError as e:
+                if e['Error']['Code'] == 'InternalFailure':
+                    self.log.error('Exception when tagging resource %s' %
+                        (['ARN'], e['Error']['Code']))
```

other errors are silently passing here.